sanic/testing.py

```diff
@@ -1,12 +1,30 @@
+import sys
 import traceback
 from json import JSONDecodeError
 from sanic.log import logger
 from sanic.exceptions import MethodNotSupported
 from sanic.response import text
 
+try:
+    try:
+        # direct use
+        import packaging as _packaging
+        version = _packaging.version
+    except (ImportError, AttributeError):
+        # setuptools v39.0 and above.
+        try:
+            from setuptools.extern import packaging as _packaging
+        except ImportError:
+            # Before setuptools v39.0
+            from pkg_resources.extern import packaging as _packaging
+    version = _packaging.version
+except ImportError:
+    raise RuntimeError("The 'packaging' library is missing.")
+
```

Comment:
We should not move this piece of code inside sanic module, it should stay in the unit tests file at least.

I'd like to refactor and get rid of this piece of code actually, i think there is one unit test relies on this because of aiohttp versions.

Reply:
this gets used in both test_keep_alive_timeout and test_request_timeout, testing seemed like the natural place to consolidated this logic. Its complicated enough that I feel like we shouldn't write it twice

Reply:
1. `test_keep_alive_timeout` and `test_request_timeout` are 2 unit tests that lives in `tests/` folder, which will not be packaged, uploaded and published. They're used for Sanic's Unit testing and Development. But when you move this code under `Sanic/`, it'll be packaged and published to pypi. I don't think this piece of code is useful to any user whoever `pip install Sanic`.
2. Although `packaging` and `setuptools` are universal tools, i don't think we should import `packaging` and `setuptools` inside Sanic modules, and they're not dependencies of Sanic. Indeed, python dependency management tools like `pip`, `pipenv` rely on `setuptools/packaging`, but tools like `poetry` don't rely on it.
3. If you think it could be reused, `tests/utils.py` probably is good place. But personally, I'd prefer to get rid of this chunk of code and refactor `test_keep_alive_timeout` and `test_request_timeout` a little bit. This piece of code is used for checking aiohttp version and handle them differently, we probably should specify a little bit more strict version constraint for aiohttp.